src/EventStore.js

```diff
@@ -50,7 +51,7 @@ class EventStore extends Store {
   _query (opts) {
     if (!opts) opts = {}
 
-    const amount = opts.limit ? (opts.limit > -1 ? opts.limit : this._index.get().length) : 1 // Return 1 if no limit is provided
+    const amount = !opts.limit || opts.limit == -1 ? this._index.get().length : opts.limit; // Return all by default. 
```

Comment:
what if opts.limit is -2

Reply:
Yeah. I doubt it would pass the test. Should we wrap it into a call to `abs`?

Reply:
it should error out imo

Reply:
@tabcat it should make no difference now. @aphelionz a test case for this is a great idea though.
So I implemented that in the [PR on orbit-db](https://github.com/orbitdb/orbit-db/pull/886).

Reply:
@aphelionz why? And if so, how?

Reply:
Implementing it that way has the advantage of being more backwards compatible,
while @aphelionz's proposal is closer to the documentation. (Since it is not specified how other negative values, besides -1 should be handled.)

I would suggest third way, that has two advantages and one disadvantage at least:
- Instead of -1, 0 is used as the value to indicate all values should be returned and is used as a default.
- The limit, that isn't 0, is taken as defining the number of values to return.
- The sign of the limit determines, whether the elements should be taken from the front (+) or back of the iterator.

Example:
```
log.iterator({ limit: 0 }).collect() == [1,2,3,4,5]
log.iterator({ limit : -2 }).collect() == [4,5]
log.iterator({ limit: 2 }).collect() == [1,2]
```

The advantage of this is that this is similar to the negative indexing in, for example, Python.

But the disadvantage is of course, that it isn't very intuitive at first. (There'll probably be many people confusing it with the `reverse` option)

Reply:
So, if you are interested in only getting the last five (in chronological order)
you provide the `limit: -5`, if you want the first five (in chronological order)
you provide the `limit: 5`. 
Of course, the chronology reverses, with the reverse option.

Reply:
Similarly if gte, gt, lt, lte is defined. 
Then it will be 5 after or before the specified hash?